bin/internal/prepare-workspace.sh

```diff
@@ -49,16 +58,23 @@ do
   component_dir=$(find_component_directory "${component_id}")
   # backward compatible purpose, some may expect this variable to be component lifecycle directory
   export LAUNCH_COMPONENT="${component_dir}/bin"
-  # FIXME: change here to read manifest `commands.validate` entry
-  validate_script=${component_dir}/bin/validate.sh
-  if [ ! -z "${component_dir}" -a -x "${validate_script}" ]; then
-    . ${validate_script}
-    retval=$?
-    let "ERRORS_FOUND=${ERRORS_FOUND}+${retval}"
+  validate_script=$(read_component_manifest "${component_dir}" ".commands.validate" 2>/dev/null)
+  if [ -z "${validate_script}" ]; then
+    # backward compatible purpose
+    print_message "unable to determine validate script from component ${component_id} manifest, fall back to default bin/validate.sh"
+    validate_script=bin/validate.sh
+  fi
+  if [ ! -z "${component_dir}" ]; then
```

Comment:
This is optional and probably fixing something that I originally wrote, but it might be worth considering `-n` here over `! -z`. When I was scanning through it I processed it as `-z` like the others and was confused for a bit.
There are a couple of other cases of this in the PR

Reply:
I'm not sure if I have enough time to run tests today before RC. Will merge and fix them in next PR really soon.